src/__e2e__/mockServer.js

```diff
@@ -0,0 +1,100 @@
+// @ts-check
+const http = require("http");
+const {
+  parseRequestParams,
+  createHandler,
+} = require("graphql-http/lib/use/http");
+const { buildSchema } = require("graphql");
+const { Trie } = require("@wry/trie");
+
+function runMockServer(
+  /** @type {number}  */ port,
+  onStart = (/** @type {number}  */ port) => {},
+) {
+  const mocks = new Trie(false);
+
+  const server = http.createServer(async (req, res) => {
+    if (req.url === "/apollo") {
```

Comment:
Would it be possible to split out some of this logic to separate functions? With all the nesting, its a bit difficult to tell what belongs to what.

i.e.

```ts
if (req.url === '/apollo' && req.method === "POST") {
  handleApolloPost(req, res);
} else if (req.url === '/apollo' && req.method === "PUT") {
  handleApolloPut(req, res);
} else if (req.url === '/graphql') {
  handleGraphQL(req, res);
} else {
  res.writeHead(200);
}

// ...

function handleApolloPost(req, res) {
  // ...
}

function handleApolloPut(req, res) {
  // ...
}

function handleGraphQL(req, res) {
  // ...
}
```

Obviously feel free to name this how you want, but I find this a bit easier to read since it flattens out a bunch of the nesting and keeps things pretty contained.